src/components/Questions.js

```diff
@@ -0,0 +1,306 @@
+
+class Questions extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            fields: [],
+            newfields: [
+                {
+                    label: '',
+                    description: '',
+                    order: null,
+                    required: false,
+                    data_type: 'char'
+                }
+            ]
+        }
+        this.onChange = this.onChange.bind(this)
+        this.onNewChange = this.onNewChange.bind(this)
+    }
+
+    async componentDidMount() {
+        this.props.getInfo()
+        const { id } = this.props.match.params
+        await this.props.getQuestions(id)
+        await this.props.getForm(id)
+    }
+
+    onChange = (e, id, str) => {
+        this.setState({
+            fields: this.state.fields.map((field, index) => {
+                return index === id ? {
+                    ...field.label = str === 'label' ? e.target.value : field.label,
+                    ...field.description = str === 'description' ? e.target.value : field.description,
+                    ...field.order = str === 'order' ? e.target.value : field.order,
+                    ...field.required = str === 'required' ? !field.required : field.required,
+                } : field
+            })
+        })
+    }
+
+    onNewChange = (e, id, str) => {
+        this.setState({
+            newfields: this.state.newfields.map((newfield, index) => {
+                return id === index ? {
+                    ...newfield.label = str === 'label' ? e.target.value : newfield.label,
+                    ...newfield.description = str === 'description' ? e.target.value : newfield.description,
+                    ...newfield.order = str === 'order' ? e.target.value : newfield.order,
+                    ...newfield.required = str === 'required' ? !newfield.required : newfield.required,
+                    ...newfield.data_type = str === 'data_type' ? e.target.value : newfield.data_type
+                } : newfield
+            })
+        })
+    }
+
+    createForm = () => {
+        return this.state.newfields.map((object, index) => (
+            <>
+            <Form key={index}>
+                <Form.Group widths='equal'>
+                    <Form.Input
+                        type='text'
+                        placeholder='Enter your question field label...'
+                        value={this.state.newfields[index].label}
+                        required
+                        onChange={(event) => this.onNewChange(event, index, 'label')}
+                    />
+                    <Form.Input
+                        control={Select}
+                        options={options}
+                        placeholder="Choose data type"
+                        required
+                        value={this.state.newfields[index].data_type}
+                        onChange={(event) => this.onNewChange(event, index, 'data_type')}
+                    />
```

Comment:
This component is causing the error that you just mentioned @bismitaguha. Here's something that will help you: https://stackoverflow.com/questions/47012169/a-component-is-changing-an-uncontrolled-input-of-type-text-to-be-controlled-erro. The error makes sense - the component is initially uncontrolled because `this.state.newfields[index].data_type` is `undefined`, and when it actually has a value - it shifts to controlled. You've to mitigate that. You've given `|| false` and `|| ''` in the next two Form Inputs, so I guess that would help.